interface/src/app/download/download.spec.js

```diff
@@ -2,21 +2,113 @@
  * Unit tests.
  */
 
-describe('DownloadCtrl', function() {
-  describe('isCurrentUrl', function() {
-    var DownloadCtrl, $location, $scope;
-
-    beforeEach(module('adage'));
-
-    beforeEach(inject(function($controller, _$location_, $rootScope) {
-      $location = _$location_;
-      $scope = $rootScope.$new();
-      DownloadCtrl = $controller('DownloadCtrl',
-                                 {$location: $location, $scope: $scope});
-    }));
-
-    it('validates DownloadCtrl controller', inject(function() {
-      expect(DownloadCtrl).toBeTruthy();
-    }));
+// Test the generic controller properties (without server calls):
+describe('DownloadCtrl w/o server calls', function() {
+  beforeEach(module('adage.download'));
+
+  var ctrl, mockService;
+  // Create a mocked "AnnotationTypes" service, which overwrites the service
+  // in download.js (because it is loaded after "adage.download" module).
+  beforeEach(module(function($provide) {
+    mockService = {
+      get: function() { }
+    };
+    $provide.value('AnnotationTypes', mockService);
+  }));
+
+  beforeEach(inject(function($controller) {
+    ctrl = $controller('DownloadCtrl');
+  }));
+
+  it('should be defined', function() {
+    expect(ctrl).toBeDefined();
```

Comment:
Since `;` is optional in JavaScript, I triggered this error by adding this line at the end of the controller definition:
```
return { };
```
Then the resulting controller is defined but not truthy.

Considering that this test case is more like a demo (this error will be detected by all the other test cases in the same describe block too), I am going to change it into:
```
  it('should be defined and truthy', function() {
    expect(ctrl).toBeDefined();
    expect(ctrl).toBeTruthy();
  });
```